### DISCUSSION TITLE
Is it possible to use existing APIs test if a uri matches a glob?

### DISCUSSION THREAD

**Discussion by ChiriVulpes** [microsoft/vscode-discussions#719] [Extension Development QnA] [answered]
There are nice utilities around globs used for VSCode extension APIs, and those internally all use the same custom glob utility, but I can't seem to find a way to use that same utility — or any wrappers around it (they don't seem to exist) — in my own extension code. (I was going to use it for a watcher in a custom `FileSystemProvider` to support `options.excludes`.) 

If it's not possible right now could the glob utility be wrapped in an official api somehow?

**Comment by isidorn** [comment, marked as answer]
The glob utility is not part of the official API. Sorry.

**Comment by phil294** [comment]
@isidorn That's unfortunate. What do you suggest extension authors should do instead? Install some 3rdparty dependency like micromatch?

**Reply by isidorn** [reply]
I think a 3rd party library is the way to go for now.
